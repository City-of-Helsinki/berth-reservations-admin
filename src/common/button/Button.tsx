import React from 'react';
import classNames from 'classnames';

import styles from './button.module.scss';

export type Props = {
  color?: 'standard' | 'brand' | 'critical' | 'secondary' | 'info';
  variant?: 'contained' | 'outlined' | 'text';
  size?: 'small' | 'standard' | 'large';
  icon?: JSX.Element;
} & React.DOMAttributes<HTMLButtonElement>;

const Button: React.SFC<Props> = ({
  children,
  color = 'standard',
  variant = 'contained',
  size = 'standard',
  icon,
  onClick,
}) => {
  return (
    <button
      type="button"
      onClick={onClick}
      className={classNames(
        styles.button,
        styles[color],
        size && styles[size],
        variant && styles[variant],
        icon && children && styles.hasIcon
      )}
    >
      {icon}
      {children}
    </button>
  );
};

export default Button;
