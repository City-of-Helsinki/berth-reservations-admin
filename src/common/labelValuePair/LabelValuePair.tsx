import React from 'react';
import classNames from 'classnames';

import styles from './labelValuePair.module.scss';

interface Props {
  label: string;
  value?: string | null | React.ReactElement;
  labelColor?: 'standard' | 'brand' | 'critical' | 'secondary';
  align?: 'left' | 'center' | 'right';
}

const LabelValuePair = ({
  label,
  value,
  labelColor = 'standard',
  align = 'left',
}: Props) => {
  return (
    <div className={styles.labelValuePair}>
      <span className={classNames(styles.label, styles[labelColor])}>
        {label}:
      </span>
      <span key="value" className={classNames(styles.value, styles[align])}>
        {value}
      </span>
    </div>
  );
};

export default LabelValuePair;
