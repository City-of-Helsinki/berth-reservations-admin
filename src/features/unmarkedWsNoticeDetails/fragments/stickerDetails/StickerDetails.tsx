import { useTranslation } from 'react-i18next';
import React from 'react';

import LabelValuePair from '../../../../common/labelValuePair/LabelValuePair';
import Section from '../../../../common/section/Section';
import ButtonWithConfirmation from '../../../../common/buttonWithConfirmation/buttonWithConfirmation';
import { formatStickerNumber } from '../../../../common/utils/format';

interface StickerDetailsProps {
  stickerNumber: number | null;
  stickerSeason: string | null;
  isAssigningNewStickerNumber?: boolean;
  handleAssignNewStickerNumber?(): void;
}

const StickerDetails = ({
  stickerNumber,
  stickerSeason,
  isAssigningNewStickerNumber,
  handleAssignNewStickerNumber,
}: StickerDetailsProps) => {
  const { t } = useTranslation();

  return (
    <Section title={t('unmarkedWsNotices.noticeDetails.stickerDetails.title')}>
      <LabelValuePair
        label={t('unmarkedWsNotices.noticeDetails.stickerDetails.number')}
        value={formatStickerNumber(stickerNumber)}
      />
      <LabelValuePair label={t('unmarkedWsNotices.noticeDetails.stickerDetails.season')} value={stickerSeason} />
      {handleAssignNewStickerNumber && (
        <ButtonWithConfirmation
          buttonText={t('unmarkedWsNotices.noticeDetails.stickerDetails.assign.buttonText')}
          modalTitle={t('unmarkedWsNotices.noticeDetails.stickerDetails.assign.modalTitle')}
          infoText={t('unmarkedWsNotices.noticeDetails.stickerDetails.assign.infoText')}
          onConfirmText={t('unmarkedWsNotices.noticeDetails.stickerDetails.assign.confirm')}
          onCancelText={t('common.cancel')}
          buttonStyle="flat"
          onConfirm={handleAssignNewStickerNumber}
          disabled={isAssigningNewStickerNumber}
        />
      )}
    </Section>
  );
};

export default StickerDetails;
