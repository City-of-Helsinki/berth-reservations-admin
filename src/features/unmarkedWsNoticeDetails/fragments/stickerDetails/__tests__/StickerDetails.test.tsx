import React from 'react';
import { mount } from 'enzyme';
import ReactModal from 'react-modal';

import StickerDetails from '../StickerDetails';
import LabelValuePair from '../../../../../common/labelValuePair/LabelValuePair';
import ButtonWithConfirmation from '../../../../../common/buttonWithConfirmation/buttonWithConfirmation';
import Button from '../../../../../common/button/Button';

describe('StickerDetails', () => {
  it('renders correctly', () => {
    const wrapper = mount(
      <StickerDetails stickerNumber={123} stickerSeason={'2020/2021'} handleAssignNewStickerNumber={jest.fn()} />
    );
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('renders leading zeros', () => {
    const wrapper = mount(<StickerDetails stickerNumber={1} stickerSeason={'2020/2021'} />);
    expect(wrapper.find(LabelValuePair).first().text()).toContain('0001');
  });

  it('renders dash when stickerNumber is null', () => {
    const wrapper = mount(<StickerDetails stickerNumber={null} stickerSeason={'2020/2021'} />);
    expect(wrapper.find(LabelValuePair).first().text()).toContain('-');
  });

  it('renders assignment button when handleAssignNewStickerNumber is defined', () => {
    const wrapper = mount(
      <StickerDetails stickerNumber={123} stickerSeason={'2020/2021'} handleAssignNewStickerNumber={jest.fn()} />
    );
    expect(wrapper.find(ButtonWithConfirmation).length).toBe(1);
  });

  it('calls handleAssignNewStickerNumber when confirming', () => {
    ReactModal.setAppElement('body');
    const handleAssignNewStickerNumberMock = jest.fn();
    const wrapper = mount(
      <StickerDetails
        stickerNumber={123}
        stickerSeason={'2020/2021'}
        handleAssignNewStickerNumber={handleAssignNewStickerNumberMock}
      />
    );
    wrapper.find(ButtonWithConfirmation).simulate('click');
    wrapper.find(Button).last().simulate('click');
    expect(handleAssignNewStickerNumberMock).toBeCalledTimes(1);
  });
});
