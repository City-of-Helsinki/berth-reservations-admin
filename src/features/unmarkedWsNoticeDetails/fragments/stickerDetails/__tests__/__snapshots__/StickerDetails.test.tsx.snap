// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StickerDetails renders correctly 1`] = `
<article
  class="section"
>
  <h4
    class="text standard h4 m title"
  >
    TARRA
  </h4>
  <section
    class="body"
  >
    <div
      class="labelValuePair"
    >
      <span
        class="label standard"
      >
        Numero:
      </span>
      <span
        class="value left"
      >
        0123
      </span>
    </div>
    <div
      class="labelValuePair"
    >
      <span
        class="label standard"
      >
        Kausi:
      </span>
      <span
        class="value left"
      >
        2020/2021
      </span>
    </div>
    <div
      class="labelValuePair"
    >
      <span
        class="label standard"
      >
        Postitettu:
      </span>
      <span
        class="value left"
      >
        -
      </span>
    </div>
    <button>
      <span
        class="text brand span"
      >
        Vaihda numero
      </span>
    </button>
  </section>
</article>
`;
